### DISCUSSION TITLE
Empty vcalendar

### DISCUSSION THREAD

**Discussion by kriegfrj** [ical4j/ical4j#574] [Q&A]
I'm trying to write a servlet that returns a ics file.

There are times when the generated caller
Calendar covers a period that doesn't have any events. Is there a way to generate ics with iCal4J that represents an empty vcalendar? From memory I tried it and got a runtime exception warning me about the empty vcalendar component.

**Comment by benfortuna** [comment]
If you use the `CalendarOutputter` class, you can specify if you want to apply validation or not:

```java
CalendarOutputter out = new CalendarOutputter(false);
out.output(calendar, writer); // where writer is the target, e.g. StringWriter, etc.
```

That should allow an empty calendar to be generated.
